### PR TITLE
Adds recipient and relayer to pallet, using new circuit.

### PR DESCRIPTION
**Pull request by drewstone**

### CODE REVIEW

**Comment by filiplazovic** [comment]
@drewstone Do a rebase or merge master, and also you can disable CircleCI for `anon` and `bulletproofs-gadgets` repo

File: pallets/merkle/src/lib.rs [hunk 1]
```diff
@@ -98,7 +98,7 @@ use bulletproofs::{
 use codec::{Decode, Encode};
 use curve25519_dalek::scalar::Scalar;
 use curve25519_gadgets::{
-	fixed_deposit_tree::fixed_deposit_tree_verif_gadget,
+	fixed_deposit_tree::{mixer_verif_gadget},
```
Comment (filiplazovic) [inline pallets/merkle/src/lib.rs:101]: Run `cargo fmt`

File: pallets/merkle/Cargo.toml [hunk 2]
```diff
@@ -36,7 +36,7 @@ default-features = false
 features = ["u64_backend", "alloc"]
 
 [dependencies.curve25519-gadgets]
-rev = "4028605"
+branch = "main"
```
Comment (filiplazovic) [inline pallets/merkle/Cargo.toml:39]: If there are breaking changes on `curve25519-gadgets`, it might not compile. Do you want it this way?

Comment (drewstone) [reply pallets/merkle/Cargo.toml:39]: I'll peg to a commit, good idea.

**Pull request merged**